### ISSUE TITLE
Bug: require-returns-check even when missing

### ISSUE THREAD

**Issue by kkmuffme** [gajus/eslint-plugin-jsdoc#892]
the require-returns-check stops at the first occurence of "return". However all functions (except void/undefined) need a return statement at the end of the function too, otherwise the function can return void.

```js
/**
 * @param {string} message
 * @return {Boolean}
 */
function abc( message ) {
    if ( message ) {
        return true;
    }
    // give an error, since technically a return is needed here, since it's not void
}
```

**Comment by brettz9** [comment]
I think I'm inclined to adapt the function as suggested (since I think it makes more sense for projects to make an explicit return of `undefined` if it is really intended), but I'm wondering whether the built-in ESLint rule [consistent-return](https://eslint.org/docs/latest/rules/consistent-return) might not be sufficient to catch this subset of cases.

**Comment by achingbrain** [comment]
I think this has started failing for code like:

```js
/**
 * @returns {string}
 */
function () {
  if (/* some condition */) {
    return 'ok'
  }

  throw new Error('aaargh')
}
```

The error is:

```
JSDoc @returns declaration present but return expression not available in function. [Error/jsdoc/require-returns-check]
```

Previously this did not cause a linting error.

**Comment by brettz9** [comment]
@achingbrain : Thanks for the report. Should now be fixed in 39.3.17. Let us know if you find any other patterns that aren't working.

**Comment by achingbrain** [comment]
Looks much better, thanks for the speed resolution!
